src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java

```diff
@@ -225,22 +266,18 @@ private Mono<TransactionActivationRequestedEvent> newTransactionActivationReques
     }
 
     private Mono<TransactionActivatedEvent> newTransactionActivatedEvent(
-                                                                         Integer amount,
-                                                                         String description,
-                                                                         String email,
+                                                                         List<PaymentRequestInfo> paymentRequestsInfo,
                                                                          String transactionId,
-                                                                         String rptId,
-                                                                         String paymentToken
+                                                                         String email
     ) {
-
         TransactionActivatedData data = new TransactionActivatedData();
         data.setEmail(email);
-        NoticeCode noticeCode = new NoticeCode(paymentToken, rptId, description, amount, null);
-        data.setNoticeCodes(Arrays.asList(noticeCode));
+        List<NoticeCode> noticeCodes = toNoticeCodeList(paymentRequestsInfo);
+        data.setNoticeCodes(noticeCodes);
```

Comment:
Pure stylistic choice, but it makes spotting missing fields in initializations easier:
```suggestion
        List<NoticeCode> noticeCodes = toNoticeCodeList(paymentRequestsInfo);
        TransactionActivatedData data = new TransactionActivatedData(
            email,
            noticeCodes,
            null,
            null
        );
```